app/src/main/java/com/cs102/studybuddy/Login.java

```diff
@@ -0,0 +1,122 @@
+package com.cs102.studybuddy;
+
+import android.content.Intent;
+import android.text.InputType;
+import android.text.TextUtils;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+import android.os.Bundle;
+import android.widget.Toast;
+
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+
+public class Login extends AppCompatActivity {
+
+    private EditText logEmail, logPassword;
+    private Button rememberMe, forgotPass, login, goSignUp;
+    private FirebaseAuth mAuth;
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        // Check if user is signed in (non-null) and update UI accordingly.
+        FirebaseUser currentUser = mAuth.getCurrentUser();
+        if (currentUser != null&&currentUser.isEmailVerified()){
+            Intent intent = new Intent(getApplicationContext(), MainActivity.class);
+            startActivity(intent);
+            finish();
+        }
+    }
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login);
+        mAuth = FirebaseAuth.getInstance();
+
+        logEmail = findViewById(R.id.email);
+        logPassword = findViewById(R.id.password);
+        rememberMe = findViewById(R.id.RememberMe);
+        forgotPass = findViewById(R.id.forgotPass);
+        login = findViewById(R.id.Login);
+        goSignUp = findViewById(R.id.goSignUp);
+
+
+        login.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String email, pass;
+                email = String.valueOf(logEmail.getText());
+                pass = String.valueOf(logPassword.getText());
+
+                if (TextUtils.isEmpty(email)){
+                    Toast.makeText(Login.this, "Enter email", Toast.LENGTH_LONG).show();
+                    return;
+                }
+                if (TextUtils.isEmpty(pass)){
+                    Toast.makeText(Login.this, "Enter email", Toast.LENGTH_LONG).show();
```

Comment:
Should say "Empty password" instead of "Empty email".

```suggestion
                    Toast.makeText(Login.this, "Enter password", Toast.LENGTH_LONG).show();
```